src/kernel/kshell.c

```diff
@@ -107,6 +107,9 @@ bool alt_mode = false;
 bool ctrl_mode = false;
 bool super_mode = false;
 
+char history_data[100][RDL_SIZE];
+uint8_t tot_cmds=0;
```

Comment:
I'd prefer you keep this in some sort of dynamic data structure, rather than a list - not just for cleanliness, but because this implementation is rife with data overflow opportunities. - for instance, you never bounds check on tot_cmds

Reply:
I have created a dynamic list to resolve the overflow problem.